### PR TITLE
Pre-C-Bindings Cleanups #3

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Hopefully this is the last of them given we're getting close on the C bindings PR. See individual commits for details.

### CODE REVIEW

File: lightning/src/routing/network_graph.rs [hunk 1]
```diff
@@ -41,6 +41,11 @@ pub struct NetworkGraph {
 	nodes: BTreeMap<PublicKey, NodeInfo>,
 }
 
+/// A simple newtype for RwLockReadGuard<'a, NetworkGraph>.
+/// This exists only to make accessing a RwLock<NetworkGraph> possible from
+/// the C bindings, as it can be done directly in Rust code.
+pub struct LockedNetworkGraph<'a>(pub RwLockReadGuard<'a, NetworkGraph>);
```
Comment (valentinewallace) [inline lightning/src/routing/network_graph.rs:47]: Hm, should we start using this in RL, just to get some test coverage on it?

Comment (TheBlueMatt) [reply lightning/src/routing/network_graph.rs:47]: Its used in the bindings `demo.cpp` :shrug:

File: lightning/src/chain/keysinterface.rs [hunk 2]
```diff
@@ -33,7 +33,7 @@ use util::ser::{Writeable, Writer, Readable};
 
 use ln::chan_utils;
 use ln::chan_utils::{HTLCOutputInCommitment, make_funding_redeemscript, ChannelPublicKeys, LocalCommitmentTransaction, PreCalculatedTxCreationKeys};
-use ln::msgs;
+use ln::msgs::UnsignedChannelAnnouncement;
```
Comment (jkczyz) [inline lightning/src/chain/keysinterface.rs:37]: Could you open an issue to fix the underlying problem in the bindings generation? Sometimes using the module prefix is preferable.

File: lightning/src/ln/channelmanager.rs [hunk 3]
```diff
@@ -3775,7 +3775,27 @@ pub struct ChannelManagerReadArgs<'a, ChanSigner: 'a + ChannelKeys, M: Deref, T:
...
+	where M::Target: ManyChannelMonitor<Keys=ChanSigner>,
+		T::Target: BroadcasterInterface,
+		K::Target: KeysInterface<ChanKeySigner = ChanSigner>,
+		F::Target: FeeEstimator,
+		L::Target: Logger,
+	{
+	/// Simple utility function to create a ChannelManagerReadArgs which creates the monitor
+	/// HashMap for you. This is primarily useful for C bindings where it is not practical to
+	/// populate a HashMap directly from C.
+	pub fn construct(keys_manager: K, fee_estimator: F, monitor: M, tx_broadcaster: T, logger: L, default_config: UserConfig,
```
Comment (jkczyz) [inline lightning/src/ln/channelmanager.rs:3792]: s/construct/new ?

File: lightning/src/chain/keysinterface.rs (hunk 2 above)
Comment (TheBlueMatt) [reply lightning/src/chain/keysinterface.rs:37]: #678

**Pull request merged**